rules/utils/quote-string.js

```diff
@@ -1,9 +1,16 @@
 'use strict';
 
 /**
-Escape apostrophe and wrap the result in single quotes.
+Escape string and wrap the result in quotes.
 
 @param {string} string - The string to be quoted.
-@returns {string} - The quoted string.
+@param {quote} string - The quote character.
+@returns {string} - The quoted and escaped string.
 */
-module.exports = string => `'${string.replace(/'/g, '\\\'')}'`;
+module.exports = (string, quote = '\'') => {
+	const escaped = string
+		.replace(new RegExp(quote, 'g'), `\\${quote}`)
```

Comment:
Are you sure this is enough? Here, I used to use a regex for escaping the string, but it could too complicated and in the end I had to use a proper parser: https://github.com/sindresorhus/to-single-quotes/commit/7f203d0f6a25ab5e74911732b77de02f8dc7eae2

Reply:
Wow, I checked the tests on that repo, seems complicated. But no one has report any issue with the old simple encode yet. I guess we can improve this if other case found.

Reply:
Found broken cases, https://github.com/sindresorhus/eslint-plugin-unicorn/pull/496/commits/b81ea17f7f5cf221dcdf214ab8e789ac18b1f4b5

Similar escape https://github.com/eslint/eslint/blob/9dfc8501fb1956c90dc11e6377b4cb38a6bea65d/lib/rules/quotes.js#L48